src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -48,12 +48,21 @@ internal static void CopyProperties(this PreferenceSettings source, PreferenceSe
     /// </summary>
     public class PreferenceSettings : NotificationObject, IPreferences, IRenderPrecisionPreference, IDisablePackageLoadingPreferences, ILogSource, IHideAutocompleteMethodOptions
     {
-        internal readonly static Lazy<PreferenceSettings>
+        private readonly static Lazy<PreferenceSettings>
             lazy = new Lazy<PreferenceSettings>
             (() => PreferenceSettings.Load(PathManager.Instance.PreferenceFilePath));
 
+        /// <summary>
+        /// Return a PreferenceSetting object.  The object returned is based on the following conditions:
+        /// 1) if DynamoModel present, the DynamoModel.PreferenceSettings object is returned,
+        /// 2) else, if a valid setting xml file exists, the PreferenceSettings object de-serialized from the xml file is returned,
+        /// 3) else, if no DynamoModel and no valid xml file exists, a new PreferenceSettings object returned
+        /// Note that Instance is a runtime object only.  No changes to the PreferenceSettings will be persisted on disk with condition 2 or 3.
+        /// User of Instance must initiate save operations to insure persistence of modifications to the PreferenceSettings model.
+        /// In some cases even the save will not guarantee persistence of modifications depending on the startup of DynamoModel.
+        /// </summary>
         [XmlIgnore]
-        public static PreferenceSettings Instance { get; internal set; } = lazy.Value;
+        internal static PreferenceSettings Instance { get; set; } = lazy.Value;
```

Comment:
The whole singleton implementation for this class is flawed. Typically in a singleton pattern, the instance should be public and the constructors should be all private.

Reply:
Unfortunately, this class has a public constructor.